File: exams/aimacana/u3/ExamStrategy/models/insertion_sort.py
from typing import List
from .sorting_strategy import SortingStrategy


class InsertionSort(SortingStrategy):

    @property
    def name(self) -> str:
        return "InsertionSort"

    def sort(self, data: List[int]) -> List[int]:
        arr = list(data)
        for i in range(1, len(arr)):
            key = arr[i]
            j = i - 1
            while j >= 0 and arr[j] > key:
                arr[j + 1] = arr[j]
                j -= 1
            arr[j + 1] = key
        return arr

File: hw/chavez/u3/hw28Strategy/Java/Strategy/src/main/java/ec/edu/espe/strategy/model/InsertionSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategy.model;

/**
 *
 * @author Odalys Chavez, CodeBreakers, @ESPE
 */
public class InsertionSort implements SortingStrategy{

    @Override
    public int[] sort(int[] data) {
        for (int i = 1; i < data.length; i++) {

            int key = data[i];
            int j = i - 1;

            while (j >= 0 && data[j] > key) {

                data[j + 1] = data[j];
                j--;
            }

            data[j + 1] = key;
        }

        return data;
    }  
}
